app/models/concerns/time_slot.rb

```diff
@@ -1,17 +1,24 @@
+require_relative './../../helpers/time_slot_helper'
+include TimeSlotHelper
+
 module TimeSlot
-  def available_time_slots(day_range, current_time)
-    slots_for_day = day_range.each_with_object({}) do |day, slots|
-      slots[day] = available_time_slots_for_day(day)
+  def available_time_slots(forward_days)
 
-      if day.today?
-        slots[day].reject! do |slot|
-          slot[:slot_start].to_i < current_time.to_i
-        end
+    day_range = []
+    d = 0
+    i = 0
+    until d == forward_days
+      day = i.days.from_now
+      if business_day?(day)
+        day_range << day
+        puts day
+        d += 1
       end
+      i += 1
     end
```

Comment:
Here is a way to filter only business days and appending in `day_range`.

This piece of code maybe not optimal for ruby language (neither pretty clean), but functional. We accepts suggestions to improve this :smiley:

Reply:
I think it's good enough for now. I only suggest giving more descriptive names for variable and extracting it to another function. In my opinion, this should be done outside `available_time_slots`, preferably. This method should work with any given `DateRange`.

Reply:
We can create a helper function `build_weekdays_date_range(number_of_days)`. It could be private inside the controller for now. Later we can extract it to a proper helper.